python/cudf/cudf/core/dtypes.py

```diff
@@ -166,3 +166,6 @@ def __repr__(self):
             return f"ListDtype({self.element_type.__repr__()})"
         else:
             return f"ListDtype({self.element_type})"
+
+    def __str__(self):
+        return "object"
```

Comment:
I'm not sure we want to do this for List dtypes. cc @shwina @brandon-b-miller

Reply:
For now, it seems like the only approach that works around what PyArrow is doing here: https://github.com/apache/arrow/blob/50d6252ea6b6692ad090e148d096a67605f13811/python/pyarrow/pandas_compat.py#L131

Reply:
I'd be okay if this was a short term fix in nightlies, but given this is right before 0.16 release this is concerning to me.

Should we just stop using the arrow function directly and reimplement it as needed while additionally adding `cudf` specific metadata? The json constructed doesn't seem overly complex.

Reply:
I encountered this building cuDF dtypes. The only workaround I could see was to construct a the metadata from `real_data.head(0).to_pandas()`. 

I don't know much about the parquet writer overall but is this change just to pacify the pandas parquet reader, and thus the tests? If pandas doesn't support lists but these parquet files can be read by other frameworks (spark, etc) I wonder how far out of our way we should go to make it work with pandas.

Reply:
> I don't know much about the parquet writer overall but is this change just to pacify the pandas parquet reader, and thus the tests? If pandas doesn't support lists but these parquet files can be read by other frameworks (spark, etc) I wonder how far out of our way we should go to make it work with pandas.

We try to capture Pandas metadata in the file so Pandas can read the data correctly as expected as well as round trip things like `RangeIndex` objects without issue. We're currently putting `list` as a `numpy_dtype` which is invalid.

Reply:
If we have to make it work with pandas, short of rewriting that pyarrow function I suppose we could monkey patch this change into those dtype instances at runtime. But I suppose what I'm asking is, does roundtripping through a library that doesn't support `list` make sense as a test for this, as opposed to some alternative (maybe pyarrow, or just cuDF itself)? I think if spark wrote a parquet file that had a list column, and then pandas was used to try and read that file, it would error right?

Reply:
> But I suppose what I'm asking is, does roundtripping through a library that doesn't support `list` make sense as a test for this, as opposed to some alternative (maybe pyarrow, or just cuDF itself)?

It is supported as an object column of lists for example. Pandas parquet reading / writing goes through PyArrow and in the case of List / Struct / Map columns PyArrow handles the translation to/from Pandas object columns as needed.

Reply:
> I think if spark wrote a parquet file that had a list column, and then pandas was used to try and read that file, it would error right?

I just checked. It works. I suppose the metadata is for additional parquet specific information like which column is the index etc. But I was able to read a parquet file that had no metadata.